utils/posts.js

```diff
@@ -0,0 +1,37 @@
+import fs from 'fs'
+import path from 'path'
+import matter from 'gray-matter'
+
+const postsDirectory = path.join(process.cwd(), 'posts')
+
+export function getSortedPostsData() {
+  // Get file names under /posts
+  const fileNames = fs.readdirSync(postsDirectory)
+  const allPostsData = fileNames.map((fileName) => {
+    // Remove ".md" from file name to get id
+    const id = fileName.replace(/\.md$/, '')
+
+    // Read markdown file as string
+    const fullPath = path.join(postsDirectory, fileName)
+    const fileContents = fs.readFileSync(fullPath, 'utf8')
+
+    // Use gray-matter to parse the post metadata section
+    const matterResult = matter(fileContents)
+
+    // Combine the data with the id
+    return {
+      id,
+      title: "Meetup Event",
+      date: "2009-01-03 00:00:00",
+      ...matterResult.data,      
```

Comment:
See here @jordan-bravo: I added a default title and date, but these get overwritten by `...matterResult.data`.
  
Another option might be to just check if date or title is empty and kill the build process, spitting out a nice error message. But this works fine too.